apps/ejabberd/src/mod_mam_utils.erl

```diff
@@ -227,38 +233,51 @@ external_binary_to_mess_id(BExtMessID) when is_binary(BExtMessID) ->
 %% -----------------------------------------------------------------------
 %% XML
 
+%% @doc Adds all arcid elements (for backward compatibility)
```

Comment:
All right, let them both be configurable then:
* `archived` - disabled by default (and warning if someone enables it with suggestion to use `stanza-id`)
* `stanza-id` - enabled by default